LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/NimbusJWTParser.java

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.labapi.utilities;
+
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A JWT Parser that uses Nimbus to parse the JWT.
+ */
+public class NimbusJWTParser implements IJWTParser {
+    @Override
+    public Map<String, ?> parseJWT(String rawJwt) {
+        try {
+            final JWT jwt = JWTParser.parse(rawJwt);
+            final JWTClaimsSet claimsSet;
+            claimsSet = jwt.getJWTClaimsSet();
+            return new HashMap<>(claimsSet.getClaims());
+        } catch (final ParseException e) {
+            throw new IllegalArgumentException(e);
```

Comment:
I'm not sure about IllegalArgument, but you should call it out in the documentation that this is what happens.

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1389/commits/bcac31bfad2e83a3efc4579b4659ecabff343454